distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/JvmErgonomics.java

```diff
@@ -53,25 +53,53 @@ private JvmErgonomics() {
      */
     static List<String> choose(final List<String> userDefinedJvmOptions) throws InterruptedException, IOException {
         final List<String> ergonomicChoices = new ArrayList<>();
-        final Map<String, Optional<String>> finalJvmOptions = finalJvmOptions(userDefinedJvmOptions);
+        final Map<String, JvmOption> finalJvmOptions = finalJvmOptions(userDefinedJvmOptions);
         final long heapSize = extractHeapSize(finalJvmOptions);
         final long maxDirectMemorySize = extractMaxDirectMemorySize(finalJvmOptions);
         if (maxDirectMemorySize == 0) {
             ergonomicChoices.add("-XX:MaxDirectMemorySize=" + heapSize / 2);
         }
+        // Use ParallelGC for heaps <= 4GB unless the user has explicitly set the garbage collector
+        JvmOption g1 = finalJvmOptions.get("UseG1GC");
+        if (heapSize <= 4 * 1024 * 1024 * 1024 && g1.getMandatoryValue().equals("true") && g1.isCommandLineOrigin() == false) {
+            ergonomicChoices.add("-XX:+UseParallelGC");
+        }
+        System.out.println("HEre inn");
         return ergonomicChoices;
     }
 
     private static final Pattern OPTION = Pattern.compile(
-        "^\\s*\\S+\\s+(?<flag>\\S+)\\s+:?=\\s+(?<value>\\S+)?\\s+\\{[^}]+?\\}\\s+\\{[^}]+}"
+        "^\\s*\\S+\\s+(?<flag>\\S+)\\s+:?=\\s+(?<value>\\S+)?\\s+\\{[^}]+?\\}\\s+\\{(?<origin>[^}]+)}"
```

Comment:
Drive-by comment, feel free to ignore - should we pass `Pattern.COMMENTS` here so that we can break up the regex with whitespace, to make it easier to read?